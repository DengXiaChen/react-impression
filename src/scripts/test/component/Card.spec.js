import React from 'react';
import { shallow } from 'enzyme';
import { Card } from '../../components/impression';

const setup = () => {
    const props = {
        block: true,
        noborder: true,
    };

    const wrapper = shallow(
        <Card {...props}>
            <Card.Header>
                header
            </Card.Header>
            <Card.Block>
                block
            </Card.Block>
            <Card.Footer>
                footer
            </Card.Footer>
        </Card>
    );

    return {
        props,
        wrapper,
    };
};

describe('Card', () => {
    const { wrapper } = setup();

    // 通过 input 是否存在来判断 Todo组件是否被渲染
    it('render', () => {
        expect(wrapper.find('.card').exists()).toBe(true);
        expect(wrapper.find('CardHeader').children().text()).toBe('header');
        expect(wrapper.find('CardBlock').children().text()).toBe('block');
        expect(wrapper.find('CardFooter').children().text()).toBe('footer');
    });

    it('block', () => {
        expect(wrapper.find('.card.card-block').exists()).toBe(true);
    });

    it('noborder', () => {
        expect(wrapper.find('.card.no-border').exists()).toBe(true);
    });

    it('ui', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
