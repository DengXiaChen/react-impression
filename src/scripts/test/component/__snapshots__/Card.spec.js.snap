// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card ui 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="card card-block no-border"
>
    <CardHeader>
        header
    </CardHeader>
    <CardBlock>
        block
    </CardBlock>
    <CardFooter>
        footer
    </CardFooter>
</div>,
  "nodes": Array [
    <div
      className="card card-block no-border"
>
      <CardHeader>
            header
      </CardHeader>
      <CardBlock>
            block
      </CardBlock>
      <CardFooter>
            footer
      </CardFooter>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Card
        block={true}
        noborder={true}
>
        <CardHeader>
                header
        </CardHeader>
        <CardBlock>
                block
        </CardBlock>
        <CardFooter>
                footer
        </CardFooter>
</Card>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Card {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "block": true,
          "children": Array [
            <CardHeader>
              header
</CardHeader>,
            <CardBlock>
              block
</CardBlock>,
            <CardFooter>
              footer
</CardFooter>,
          ],
          "noborder": true,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="card card-block no-border"
>
          <CardHeader>
                    header
          </CardHeader>
          <CardBlock>
                    block
          </CardBlock>
          <CardFooter>
                    footer
          </CardFooter>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="card card-block no-border"
>
          <CardHeader>
                    header
          </CardHeader>
          <CardBlock>
                    block
          </CardBlock>
          <CardFooter>
                    footer
          </CardFooter>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Card
    block={true}
    noborder={true}
>
    <CardHeader>
        header
    </CardHeader>
    <CardBlock>
        block
    </CardBlock>
    <CardFooter>
        footer
    </CardFooter>
</Card>,
}
`;
